contracts/extensions/ETHOrderClonable.sol

```diff
@@ -0,0 +1,225 @@
+        thisIsClone
+        onlyByFactory
+    {
+        _WETH.safeDeposit(msg.value);
+        _WETH.forceApprove(_LIMIT_ORDER_PROTOCOL, msg.value);
+    }
+
+    function isValidSignature(bytes32 hash, bytes calldata signature)
+        external
+        view
+        thisIsClone
+        returns(bytes4)
+    {
+        if (signature.length != _ORDER_AND_TRAITS_BYTES_LENGTH) revert SignatureShouldContainOrderAndTraitsButLengthIsWrong(_ORDER_AND_TRAITS_BYTES_LENGTH, signature.length);
+        OrderAndTraits calldata orderAndTraits;
+        assembly ("memory-safe") { // solhint-disable-line no-inline-assembly
+            orderAndTraits := signature.offset
+        }
+
+        // Check order args by CREATE2 salt validation
+        address maker = orderAndTraits.order.maker.get();
+        bytes32 origOrderHash = orderAndTraits.order.hash(_domainSeparatorV4());
+        address clone = addressOf(maker, origOrderHash, orderAndTraits.auction);
+        if (clone != address(this)) {
+            return bytes4(0);
+        }
+
+        // Check if patched order from signature matches LOP filling order
+        IOrderMixin.Order memory order = orderAndTraits.order;
+        order.maker = Address.wrap(uint160(address(this)));
+        bytes32 orderHash = order.hashMemory(_domainSeparatorV4());
+        if (orderHash != hash) {
+            return bytes4(0);
+        }
+
+        return this.isValidSignature.selector;
+    }
+
+    function cancelOrder(bytes32 orderHash, Auction auction) external thisIsClone {
+        _cancelOrder(msg.sender, orderHash, auction, 0);
+    }
+
+    function cancelExpiredOrderByResolver(address maker, IOrderMixin.Order calldata origOrder, Auction auction)
+        external
+        thisIsClone
+        onlyResolver
+    {
+        uint256 orderExpirationTime = origOrder.makerTraits.getExpirationTime();
+        if (orderExpirationTime > 0 && block.timestamp > orderExpirationTime) revert OrderShouldBeExpired(block.timestamp, orderExpirationTime);
+
+        // Using EIP712Alien._domainSeparatorV4() is cheaper than call IOrderMixin(_LIMIT_ORDER_PROTOCOL).hashOrder(order);
+        bytes32 origOrderHash = origOrder.hash(_domainSeparatorV4());
+        uint256 resolverReward = getResolverReward(auction, orderExpirationTime);
+        _cancelOrder(maker, origOrderHash, auction, resolverReward);
+    }
+
+    function _cancelOrder(address maker, bytes32 origOrderHash, Auction auction, uint256 resolverReward)
+        private
+        thisIsCloneFor(maker, origOrderHash, auction)
+    {
+        uint256 balance = _WETH.safeBalanceOf(address(this));
+        if (balance == 0) revert CanNotCancelForZeroBalance();
+
+        if (resolverReward > 0) {
+            balance -= resolverReward;
+            _WETH.safeWithdrawTo(resolverReward, msg.sender);
+        }
+        if (balance > 0) {
+            _WETH.safeWithdrawTo(balance, maker);
+        }
+        emit CloneCancelled(maker, origOrderHash, balance);
```

Comment:
Backend need to have two separate Events for cancel by user and cancel by resolver

Reply:
Fixed in 8f9f3b0741fb07fbc01e42e22af13869eec76470